Exporter/Exporter.php

```diff
@@ -14,13 +14,53 @@
 use Exporter\Source\SourceIteratorInterface;
 use Exporter\Writer\CsvWriter;
 use Exporter\Writer\JsonWriter;
+use Exporter\Writer\WriterInterface;
 use Exporter\Writer\XlsWriter;
 use Exporter\Writer\XmlWriter;
 use Symfony\Component\HttpFoundation\StreamedResponse;
 
 class Exporter
 {
     /**
+     * @var array[string => WriterInterface]
+     */
+    private $writers;
+
+    /**
+     * NEXT MAJOR: remove default value.
+     *
+     * @param array[string => WriterInterface] writers a hash of allowed writers
```

Comment:
Should be `@param WriterInterace[] ...`

AFAIK your notation is not offical supported and doesn't work with autocompletion.